apps/gdalwarp_lib.cpp

```diff
@@ -5754,11 +5754,20 @@ GDALWarpAppOptionsGetParser(GDALWarpAppOptions *psOptions,
         .action(
             [psOptions](const std::string &s)
             {
-                if (CPLAtofM(s.c_str()) < 10000)
+                if (EQUAL("unlimited", s.c_str()))
+                {
+                    psOptions->dfWarpMemoryLimit =
```

Comment:
FYI, we do have a CPLGetUsablePhysicalRAM() . Taking 80% of it might be a way to get a similar effect to unlimited without crashing. Although maybe 80% is too much because if you have a warp working buffer of that size, you'll need an input and output buffers of comparable size. So perhaps 25% would be the actual practical limit.

Reply:
I get a graceful exit if I don't have enough memory available: `ERROR 2: /home/dan/dev/gdal/alg/gdalwarpoperation.cpp, 694: cannot allocate 777600000000 bytes`

Would `-wm x%` be better? So I can still write `-wm 100%` if I want to live on the edge?

Reply:
> I get a graceful exit if I don't have enough memory available: `ERROR 2: /home/dan/dev/gdal/alg/gdalwarpoperation.cpp, 694: cannot allocate 777600000000 bytes`

yes, because you are requesting way too much :-) There might be situation where you require almost too much, and might get caught by the OOM killer.

Anyway, I don't call for changes. "Unlimited" is pretty clear. This was a FYI that we have some infrastructure to get an idea of what is available (the name might be a bit misleading "CPLGetUsablePhysicalRAM". This is what the process can  use in total,  not necessarily what is remaining, and I wouldn't exclude that the OS might adveritze more memory than the OOM killer will really allow)

Reply:
People tend to believe that for full speed 100% of anything is the best. If I understand right, in this case it may not be true. Should there a comment about that if there isn't already?

Reply:
> Should there a comment about that if there isn't already?

There's some discussion at https://gdal.org/en/latest/programs/gdalwarp.html#memory-usage

Reply:
> https://gdal.org/en/latest/programs/gdalwarp.html#memory-usage

ah indeed, but might be good to link to it / add some short warning very clause to the mention of unlimited that will make people extremely excited ;-)
Or maybe ... if it is just an experimental flag ... just not document it .... ?